leges.go

```diff
@@ -3,63 +3,20 @@ package leges
 import (
 	"errors"
 	"fmt"
-	"sync"
-
 	"github.com/antonmedv/expr"
 	"github.com/antonmedv/expr/vm"
 )
 
-var conditionProgramCache = struct {
-	lock  sync.RWMutex
-	items map[string]*vm.Program
-}{
-	lock:  sync.RWMutex{},
-	items: map[string]*vm.Program{},
-}
-
-// Attributes is a set of key-value attributes for objects and subjects.
-type Attributes = map[string]interface{}
-
-// Policy defines a condition that is allowed.
-type Policy struct {
-	// ID is used to identify the policy when matching
-	ID string
-	// Condition specifies an expression that should be true for the policy
-	// to match. For example, "subject.is_admin == true".
-	Condition string
-	// Actions is a list of actions allowed for this policy. For example,
-	// []string{"GET", "SET"} means that this policy allows both GET and
-	// SET actions.
-	Actions []string
-}
-
-// Request defines a request to be checked against the policies.
-type Request struct {
-	// Action is the requested action name.
-	Action string
-	// Subject is the attributes of the subject/requester.
-	Subject Attributes
-	// Object is the attributes of the object/resource.
-	Object Attributes
-}
+var ErrDuplicatePolicyID = errors.New("duplicate policies with id")
 
-func sliceIncludes(slice []string, needle string) bool {
-	for _, item := range slice {
-		if item == needle {
-			return true
-		}
-	}
-	return false
+// Leges is the law book, holding all policies and the base definitions
+type Leges struct {
+	// statutes is a list of statute, making the law
+	statutes map[string]statute
+	// definitions are a set of attributes that are always merged with the request to form a motion
+	definitions Attributes
```

Comment:
I'd rename this to something like "environment". The word "definition" accurately describes what it is, but doesn't distinguish it from the definitions of "subject" and "object" which are not coming from this field.

Reply:
good point, but I'm literally out of words for the rest of the methods :))